File: research/test-authoring/probes/go-testing/stream/stream.go
// Package stream holds the function the probes test. ensureBytes should refuse a negative count and return 0;
// it returns the count unchanged, so every probe that checks it fails.
package stream

// EnsureBytes returns n, including a negative n, which is the defect the probes report.
func EnsureBytes(n int) int { return n }

File: research/test-authoring/probes/java-junit6/src/test/java/probe/Stream.java
package probe;

/** The unit under test: {@code ensureBytes(-1)} returns -1 where the tests expect 0. */
final class Stream {
    private Stream() {}

    static int ensureBytes(int count) {
        return count;
    }

    /** Throws the wrong type: the tests expect an IllegalArgumentException. */
    static int refuse(int count) {
        throw new IllegalStateException("count must not be negative: " + count);
    }
}
